queues/c-or-cpp/queue-linked-list.cpp

```diff
@@ -0,0 +1,119 @@
+// Content: Queue using linked list (FIFO)
+// Author: Tawfik Yasser
+// Date: Fri, 5 Mar 2021
+
+#include <iostream>
+using namespace std;
+// Class `Node` to represent every item in the queue
+class Node
+{
+public:
+    int data;    // Item data
+    Node *next;  // Pointer to next item
+    Node() {}    // Empty constructor
+};
+// Claas to represent the queue using linked list
+class QueueLL
+{
+private:
+    Node *front; // Pointer to the head of the queue
+    Node *rear;  // Pointer to the end of the queue
+    int _length;
+public:
+    QueueLL() //Empty constructor to initialize front and rear to NULL because in the start the queue will be empty
+    {
+        front = rear = NULL;
+    }
+
+    // Function `Enqueue` to add new item to the queue from the rear side (End)
+    void enqueue(int value)
+    {
+        Node *newNode = new Node();
+        newNode->data = value;
+        newNode->next = NULL;
+        if(front == NULL && rear == NULL) // Empty queue
+        {
+            front = rear = newNode;
+        }
+        else
+        {
+            rear->next = newNode;
+            rear = newNode;
+        }
+    }
+ 
+    // Function `Dequeue` to remove an item from the queue from the front side (Start)
+    void dequeue()
+    {
+        Node *temp = front;
+        front = front->next;
+        delete temp; //Free the memory
+    }
+
+    // Function `Peek` to return the first item in the queue
+    void peek()
+    {
+        cout<<front->data<<endl;
+    }
+
+    // Function `Print` to display the items of the queue
+    void print()
+    {
+        if(front == NULL && rear == NULL) //Empty queue
+        {
+            cout<<"Empty Queue"<<endl;
+        }
+        else
+        {
+            Node *temp = front;
+            while(temp != NULL) // Till the end of the queue
+            {
+                cout<<"[ "<<temp->data<<" ] ";
+                temp = temp->next;
+            }
+        }
+    }
+
+
+    // Function to calculate the size of the queue (The actual elments in the queue)
+    void calcSize()
```

Comment:
@TawfikYasser This function is not really needed. You can keep track of the length of queue with each addition or deletion of element. For eg. you can initialize `_length = 0` in the constructor and then in `enqueue` function do `_length++` and in the `dequeue` function do `_length--`

Reply:
Thanks @atin I will do that

Reply:
Changes done